gm4_mountaineering/data/gm4_mountaineering/functions/ski_effects/poles_jump_boost.mcfunction

```diff
@@ -0,0 +1,15 @@
+# Give jump boost effects when holding poles, with a cooldown
+# @s = player holding poles
+# at @s
+# run from main
+
+# Add jump boost effect (1 regularly, 2 if having an upwards velocity)
+execute if score @s gm4_mountaineering_poles_jump_boost matches ..0 if score @s gm4_mountaineering_vy matches 2.. run effect give @s jump_boost 1 1 true
+execute if score @s gm4_mountaineering_poles_jump_boost matches ..0 if score @s gm4_mountaineering_vy matches ..0 run effect give @s jump_boost 1 0 true
+
+# Reduce jump cooldown
+execute if score @s gm4_mountaineering_poles_jump_boost matches 1.. run scoreboard players remove @s gm4_mountaineering_poles_jump_boost 1
+
+# If jump, reset cooldown and jump counter
+execute if entity @s[scores={gm4_mountaineering_jump_counter=1..}] run scoreboard players set @s gm4_mountaineering_poles_jump_boost 3
+execute if entity @s[scores={gm4_mountaineering_jump_counter=1..}] run scoreboard players reset @s gm4_mountaineering_jump_counter
```

Comment:
I'm not sure about the whole cooldown aspect of this, it felt kind of awkward to use in-game, especially getting jump boost II was rare as to climb and get the boost you had to jump which put it on cooldown.

Reply:
The reason for the cooldown is because, even with poles, I don't see them as a tool for making many jumps with - only occasionally. Hence the desire to prevent them being used as such.

The jump boost II use case was made more with the intention of a ski-jumping-ramp (e.g. slabs or snow layers) to allow you to get a large upward velocity, then 'launch' yourself off that. I can get rid of it if it's annoying, but at least the reasoning is clear.

(I've also added some extra poles functionality, which I'll summarise in the main post later)